src/palace/manager/integration/license/opds/odl/importer.py

```diff
@@ -1,347 +1,37 @@
-
-from pydantic import TypeAdapter, ValidationError
-from sqlalchemy.orm import Session
+from pydantic import TypeAdapter
 
 from palace.manager.core.exceptions import PalaceValueError
-from palace.manager.data_layer.bibliographic import BibliographicData
-from palace.manager.data_layer.circulation import CirculationData
-from palace.manager.data_layer.identifier import IdentifierData
 from palace.manager.integration.base import integration_settings_load
+from palace.manager.integration.license.opds.importer import OpdsImporter
 from palace.manager.integration.license.opds.odl.api import OPDS2WithODLApi
 from palace.manager.integration.license.opds.odl.extractor import OPDS2WithODLExtractor
-from palace.manager.integration.license.opds.opds2.settings import OPDS2ImporterSettings
 from palace.manager.integration.license.opds.requests import (
-    BaseOpdsHttpRequest,
     OpdsAuthType,
     get_opds_requests,
 )
-from palace.manager.opds import opds2, rwpm
-from palace.manager.opds.odl import odl
-from palace.manager.opds.odl.info import LicenseInfo
 from palace.manager.opds.odl.odl import Opds2OrOpds2WithOdlPublication
 from palace.manager.opds.opds2 import PublicationFeedNoValidation
 from palace.manager.service.integration_registry.license_providers import (
     LicenseProvidersRegistry,
 )
-from palace.manager.service.redis.models.set import IdentifierSet
 from palace.manager.sqlalchemy.model.collection import Collection
-from palace.manager.util.http import HTTP, BadResponseException
-from palace.manager.util.log import LoggerMixin
-
-
-class ApplyBibliographicCallable(Protocol):
-    """
-    A callable that applies bibliographic data to the system.
-
-    For example, this could be the signature of a Celery task that processes
-    bibliographic data and updates the database accordingly:
-    apply.bibliographic_apply.delay
-    """
-
-    def __call__(
-        self, bibliographic: BibliographicData, /, *, collection_id: int
-    ) -> Any: ...
-
-
-class ApplyCirculationCallable(Protocol):
-    """
-    A callable that applies circulation data to the system.
-
-    For example, this could be the signature of a Celery task that processes
-    circulation data and updates the database accordingly:
-    apply.circulation_apply.delay
-    """
-
-    def __call__(
-        self, circulation: CirculationData, /, *, collection_id: int
-    ) -> Any: ...
-
-
-class ImporterSettingsProtocol(Protocol):
-    """
-    A protocol that defines the settings required for an OPDS2WithODLImporter.
-    """
-
-    @property
-    def external_account_id(self) -> str: ...
-    @property
-    def ignored_identifier_types(self) -> list[str]: ...
-    @property
-    def custom_accept_header(self) -> str: ...
-
-
-PublicationType = TypeVar("PublicationType", bound=opds2.BasePublication)
-SettingsType = TypeVar("SettingsType", bound=ImporterSettingsProtocol)
-
-
-class OPDS2WithODLImporter(Generic[PublicationType, SettingsType], LoggerMixin):
```

Comment:
This class mostly became `OpdsImporter`, with some refactoring to allow it to be generic enough to handle both OPDS1 and OPDS2 content.